backend/app/routes/admin/index.js

```diff
@@ -3,12 +3,18 @@ const validationMiddleware = require('../../../helpers/middlewares/validation');
 const postSuperAdminSchema = require('./@validationSchema');
 const createSuperAdmin = require('./createSuperAdmin');
 const postSuperAdmin = require('./postSuperAdmin');
+const postAdmin = require('./postAdmin');
 
 router.get('/createSuperAdmin', createSuperAdmin);
 router.post(
   '/superAdmin',
   validationMiddleware(postSuperAdminSchema),
   postSuperAdmin
 );
+router.post(
+  '/admin',
+  validationMiddleware(postSuperAdminSchema),
```

Comment:
This will be an authorised API. Only existing superAdmins can add new admins

Reply:
Sir so please can you suggest what should I do , remove the use of this mentioned middleware and make another one or what?

Reply:
Go through the auth module. You need to use the auth middleware also. Moreover, during the post request, you need to check if the request is coming from a superAdmin or not

Reply:
ok sir